Sources/GRPC/AsyncAwaitSupport/NIOAsyncWrappers.swift

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import NIOCore
+
+/// Unchecked-sendable wrapper for ``NIOAsyncWriter/Sink``, to avoid getting sendability warnings.
+@available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+struct AsyncSink<Element: Sendable>: @unchecked Sendable {
+  private let sink: NIOAsyncWriter<
+    Element,
+    GRPCAsyncWriterSinkDelegate<Element>
+  >.Sink
+
+  public init(wrapping sink: NIOAsyncWriter<
```

Comment:
This shouldn't be public and it should be `@inlinable`

Reply:
It shouldn't be public to avoid exposing all of the generic parameters for the NIOAsyncWriter type? Is it okay that the init isn't public but all of the other methods are? Or should those also not be public?

Reply:
I don't think any method here should be public since nobody except us in the internals are using this type. Or am I seeing something wrong?

Reply:
Yeah no, you're right, it was me who got confused with another type. Makes sense.